linera-base/src/data_types.rs

```diff
@@ -1064,23 +1138,34 @@ pub struct Blob {
 
 impl Blob {
     /// Creates a new [`Blob`] from the provided bytes and [`BlobId`]. Does not check if the bytes match the ID!
-    pub fn new_with_id_unchecked(id: BlobId, bytes: Vec<u8>) -> Self {
-        BlobContent::new_with_id_unchecked(id, bytes).into()
+    pub fn new_with_id_unchecked(id: BlobId, bytes: Vec<u8>) -> Result<Self, bcs::Error> {
+        BlobContent::new_with_id_unchecked(id, bytes)?.try_into()
     }
 
     /// Creates a new data [`Blob`] from the provided bytes.
-    pub fn new_data(bytes: Vec<u8>) -> Self {
-        BlobContent::new_data(bytes).into()
+    pub fn new_data(bytes: Vec<u8>) -> Result<Self, bcs::Error> {
+        BlobContent::new_data(bytes).try_into()
```

Comment:
That really can't fail, can it? Would it make sense to `expect` here?

Reply:
Yes, if it fails, something is very wrong 😅 so it might make sense to panic. I'll change it!